### ISSUE TITLE
[question]When the matched image has a large angle of rotation, the matching effect is relatively poor.

### ISSUE THREAD

**Issue by Martin0Li** [facebookresearch/silk#8]
Thank you very much for your excellent work.
I have a question about image rotation.
Due to the lack of rotational invariance in convolution, AI matching performance on images with large angle of rotation(more than 180 degree), such as those taken from drones, is often worse than traditional algorithms like SIFT, ORB, and others. In the face of this problem, do you have any ideas for improvement? I am looking forward to your response.

**Comment by mertkaraoglu** [comment]
Hi Martin, there is a great line of work from academics such as [Erik Bekkers](https://ebekkers.github.io/), [Max Welling](https://scholar.google.com/citations?user=8200InoAAAAJ&hl=en) and many more that study rotational equivariance in CNNs.

My recommendation for you is checking out a relatively recent work [SE2-LoFTR](https://github.com/georg-bn/se2-loftr) that extends LoFTR with steerable convolutions. From there you can navigate yourself to more fundamental papers in the domain.

**Comment by Martin0Li** [comment]
> 
Thank you so much, it is a great help to me.

**Comment by gleize** [comment]
Hi @Martin0Li ,

I'm not too surprised this breaks under large in-plane rotations. Since most visual odometry tasks do not have large in-plane rotations (i.e. sky is up, ground is down), we followed SuperPoint in setting small z-rotation range when sampling random homographies (between $-\frac{\pi}{8}$ to $+\frac{\pi}{8}$ to be precise, defined [here](https://github.com/facebookresearch/silk/blob/09ded25ec7673ce2084996a2376ccf7b54e1c87d/etc/models/silk-rand-homo.yaml)).

However, the direction @mertkaraoglu is mentioning has a good chance of working if you use those steerable CNNs and increase the rotation range I mentioned above.

Another alternative (untested and maybe expensive) would be to rotate the target image N times, perform the matching N times, and select the optimal rotation (based on a metric like # of matches, # of inliers or average matched similarities). It has the advantage of using a pre-trained model, but the expensive nature of the algorithm might not be suitable for your use case.

**Comment by Crazylov3** [comment]
I just saw your config about random homography. I wonder why you don't use translation. Is there any reason behind this?

**Comment by gleize** [comment]
Hi @Crazylov3,

> I wonder why you don't use translation. Is there any reason behind this?

If we shift the image by an integer value, it will simply shift the features and the correspondences, but it will not change the loss.

There is an argument to be made about shifting the image by sampling from float value in $[0,1[$ (i.e. sub-pixel translation), but we haven't tried that.
